diff --git a/dev/vs/language/typescript/tsWorker.js b/dev/vs/language/typescript/tsWorker.js
index cb194f17086a8fa0c6c15a125f3403af1a8e106f..7d6042767cc5a09262bd2ad9e313b2f252b28e3e 100644
--- a/dev/vs/language/typescript/tsWorker.js
+++ b/dev/vs/language/typescript/tsWorker.js
@@ -19471,7 +19471,7 @@ ${lanes.join("\n")}
         init_ts2();
         resolvingEmptyArray = [];
         externalHelpersModuleNameText = "tslib";
-        defaultMaximumTruncationLength = 160;
+        defaultMaximumTruncationLength = 1600;
         noTruncationMaximumTruncationLength = 1e6;
         stringWriter = createSingleLineStringWriter();
         getScriptTargetFeatures = /* @__PURE__ */ memoize(
diff --git a/esm/vs/language/typescript/ts.worker.js b/esm/vs/language/typescript/ts.worker.js
index 2858333c5e9fe2cf3e59848c966be36031afac3d..549429bc362fdb86c4be8c15d77ccc7881d526a0 100644
--- a/esm/vs/language/typescript/ts.worker.js
+++ b/esm/vs/language/typescript/ts.worker.js
@@ -19453,7 +19453,7 @@ ${lanes.join("\n")}
       init_ts2();
       resolvingEmptyArray = [];
       externalHelpersModuleNameText = "tslib";
-      defaultMaximumTruncationLength = 160;
+      defaultMaximumTruncationLength = 1600;
       noTruncationMaximumTruncationLength = 1e6;
       stringWriter = createSingleLineStringWriter();
       getScriptTargetFeatures = /* @__PURE__ */ memoize(
