diff --git a/dist/index.js b/dist/index.js
index 4b5eff9ae46b5fa0af75b16ef21f5b93142a7251..bc1651dfd82210b69ec9b5709c5f6a6320ca31b0 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -14,9 +14,9 @@ const r = /* @__PURE__ */ new Set(), s = (n, e) => e.some((t) => typeof t == "st
     if (r.size === 0)
       return e;
     const t = "@id/", u = new RegExp(
-      `${n}${t}(${[...r].join(
+      `${n}${t}(${[...r].map(it => it.replace(/\?/g, "\\?")).join(
         "|"
-      )})`,
+      )})(?:\\?external)?`,
       "g"
     );
     return u.test(e) ? e.replace(
